cou/apps/auxiliary.py

```diff
@@ -183,6 +183,92 @@ class RabbitMQServer(AuxiliaryApplication):
     # COU changes to 3.9 if the channel is set to 3.8
     multiple_channels = True
 
+    def upgrade_plan_sanity_checks(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> None:
+        """Run sanity checks before generating upgrade plan.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan, defaults to None
+        :type units: Optional[list[Unit]], optional
+        :raises ApplicationError: When application is wrongly configured.
+        :raises HaltUpgradePlanGeneration: When the application halt the upgrade plan generation.
+        :raises MismatchedOpenStackVersions: When the units of the app are running
+                                             different OpenStack versions.
+        """
+        self._check_channel()
+        self._check_application_target(target)
+        self._check_mismatched_versions(units)
+        logger.info(
+            "%s application met all the necessary prerequisites to generate the upgrade plan",
+            self.name,
+        )
+
+    def pre_upgrade_steps(
+        self, target: OpenStackRelease, units: Optional[list[Unit]]
+    ) -> list[PreUpgradeStep]:
+        """Pre Upgrade steps planning.
+
+        :param target: OpenStack release as target to upgrade.
+        :type target: OpenStackRelease
+        :param units: Units to generate upgrade plan
+        :type units: Optional[list[Unit]]
+        :return: List of pre upgrade steps.
+        :rtype: list[PreUpgradeStep]
+        """
+        steps = super().pre_upgrade_steps(target, units)
+        if self.config["enable-auto-restarts"].get("value") is False:
+            units_to_run_action = self.units.values() if units is None else units
+            steps += [
+                PreUpgradeStep(
+                    description=(
+                        "Enable auto restarts is disabled, : will"
```

Comment:
Why there is `, :`?
not-blocking comment, we should try to make description as short as possible, since in CLI it can be hard to read if it's multi line.

Reply:
`, :` is a typo, thanks! and I will try to make it shorter